libretroshare/src/services/p3discovery2.cc

```diff
@@ -243,87 +236,80 @@ void p3discovery2::removeFriend(const SSLID &sslId)
 	}
 }
 
-
 PGPID p3discovery2::getPGPId(const SSLID &id)
 {
 	PGPID pgpId;
 	mPeerMgr->getGpgId(id, pgpId);
 	return pgpId;
 }
 
-
 int p3discovery2::tick()
 {
 	return handleIncoming();
 }
 
 int p3discovery2::handleIncoming()
 {
-	RsItem *item = NULL;
+	RsItem* item = nullptr;
 
 	int nhandled = 0;
 	// While messages read
 	while(nullptr != (item = recvItem()))
 	{
-		RsDiscPgpListItem *pgplist = nullptr;
-		RsDiscPgpCertItem *pgpcert = nullptr;
-		RsDiscContactItem *contact = nullptr;
-		RsDiscIdentityListItem *gxsidlst = nullptr;
-		nhandled++;
+		RsDiscPgpListItem* pgplist = nullptr;
+		RsDiscPgpCertItem* pgpcert = nullptr;
+		RsDiscContactItem* contact = nullptr;
+		RsDiscIdentityListItem* gxsidlst = nullptr;
+		RsGossipDiscoveryInviteItem* invite = nullptr;
+		RsGossipDiscoveryInviteRequestItem* inviteReq = nullptr;
+
+		++nhandled;
 
 #ifdef P3DISC_DEBUG
 		std::cerr << "p3discovery2::handleIncoming() Received Message!" << std::endl;
 		item -> print(std::cerr);
 		std::cerr  << std::endl;
 #endif
 
-		if (nullptr != (contact = dynamic_cast<RsDiscContactItem *> (item)))
+		if((contact = dynamic_cast<RsDiscContactItem *>(item)))
```

Comment:
Well, Phenom will probably complain about this, and I quite agree with him for once: this syntaxe is very misleading. The NULL != (a=b)  was here on purpose. The compiler will simplify it for you.

Reply:
It just make the line longer and add a negation to the condition so it take a little bit more time to understanding what is happening, but I promise I will not change it in all the code :angel: just this time that I had to read it a lot because I was making big changes.

Reply:
Now that you understand it and do not need to re-write it to understand it, please revert that to the safer version. I bet that someone in the future will replace the = by a == without the person responsible for proof readign the PR being able to notice it.